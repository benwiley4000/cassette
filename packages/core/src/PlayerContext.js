import createSingleGlobalContext from './utils/createSingleGlobalContext';

// TODO: test to make sure context contents stay in sync with
// enumerated list here

export default createSingleGlobalContext({
  displayName: 'PlayerContext',
  keysWillUpdate: [
    'playlist',
    'activeTrackIndex',
    'trackLoading',
    'paused',
    'currentTime',
    'seekPreviewTime',
    'seekInProgress',
    'awaitingPlayResume',
    'duration',
    'bufferedRanges',
    'playedRanges',
    'seekableRanges',
    'volume',
    'muted',
    'shuffle',
    'stalled',
    'playbackRate',
    'setVolumeInProgress',
    'repeatStrategy',
    'mediaCannotPlay'
  ]
});
