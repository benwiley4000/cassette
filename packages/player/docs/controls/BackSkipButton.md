```jsx
import { BackSkipButton } from '@cassette/player';

<div
  style={{
    background: '#333',
    borderRadius: 3,
    display: 'inline-block'
  }}
>
  <BackSkipButton />
</div>;
```
