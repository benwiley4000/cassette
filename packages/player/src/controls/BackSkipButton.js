import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import { playerContextFilter } from '@cassette/core';

import SkipButton from './common/SkipButton';

/**
 * A button which, when clicked, either skips to the previous track in the playlist or to the beginning of the current playing track, depending upon the current elapsed time
 */
export class BackSkipButton extends PureComponent {
  render() {
    return <SkipButton back onClick={this.props.onBackSkip} />;
  }
}

BackSkipButton.propTypes = {
  onBackSkip: PropTypes.func.isRequired
};

export default playerContextFilter(BackSkipButton, ['onBackSkip']);
