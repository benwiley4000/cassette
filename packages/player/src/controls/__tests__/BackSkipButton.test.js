import React from 'react';
import renderer from 'react-test-renderer';

import PlayerContext from '@cassette/core/src/PlayerContext';

import BackSkipButton from '../BackSkipButton';

const mockPlayerContext = {
  onBackSkip: () => null
};

test('Back skip button renders correctly', () => {
  const tree = renderer
    .create(
      <PlayerContext.Provider value={mockPlayerContext}>
        <BackSkipButton />
      </PlayerContext.Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
