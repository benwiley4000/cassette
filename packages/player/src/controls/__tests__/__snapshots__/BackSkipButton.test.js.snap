// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Back skip button renders correctly 1`] = `
<div
  className="cassette__media_button_wrapper"
>
  <button
    className="cassette__skip_button cassette__media_button back"
    onClick={[Function]}
    type="button"
  >
    <div
      className="skip_button_inner foreground"
    >
      <div
        className="right_facing_triangle"
      />
      <div
        className="right_facing_triangle"
      />
      <div
        className="vertical_bar"
      />
    </div>
  </button>
</div>
`;
